docs/storage-architecture.md

```diff
@@ -0,0 +1,189 @@
+│  - Handles encryption/decryption                            │
+│  - Manages business logic                                   │
+└────────────┬──────────────┬──────────────┬──────────────────┘
+             │              │              │
+             ▼              ▼              ▼
+    ┌────────────┐  ┌──────────────┐  ┌──────────────┐
+    │     DB     │  │      FS      │  │   Blossom    │
+    └────────────┘  └──────────────┘  └──────────────┘
+    database/       storage/          BlossomClient
+    media_files     media_files       (nostr_sdk)
+```
+
+## Layer Responsibilities
+
+### 1. `whitenoise/media_files.rs` (Orchestration Layer)
+**Purpose:** High-level coordination between storage and database
+
+**Responsibilities:**
+- Coordinate between storage layer and database layer
+- Provide convenience methods for common operations
+- Combine filesystem and database operations atomically
+- Business logic and validation
+
+**Does NOT:**
+- Touch the filesystem directly (delegates to storage layer)
+- Execute SQL directly (delegates to database layer)
+- Handle network operations (use BlossomClient)
+- Handle encryption/decryption (MDK's responsibility)
+
+**Key Types:**
+- `MediaFiles<'a>` - Orchestrator with lifetime-bound references to storage and database
+
+**Key Methods:**
+- `store_and_record()` - Store file and record metadata in one operation
+- `record_in_database()` - Record existing file metadata
+- `find_file_with_prefix()` - Delegate to storage layer
+- `get_file_path()` - Delegate to storage layer
+
+### 2. `database/media_files.rs` (Left Branch)
+**Purpose:** Database operations for media file metadata
+
+**Responsibilities:**
+- Store and retrieve media file records
+- Track file metadata (hash, mime type, dimensions, blurhash)
+- Update accessed_at timestamps
+- Query files by group and hash
+
+**Does NOT:**
+- Touch the filesystem
+- Download or upload files
+- Handle encryption/decryption
+
+**Key Types:**
+- `MediaFile` - Database record representation
+- `MediaFileParams` - Parameters for saving records
+- `MediaFileRow` - Internal database row type
+
+### 3. `storage/media_files.rs` (Middle Branch)
+**Purpose:** Filesystem operations for media files
+
+**Responsibilities:**
+- Create directory structures (`<cache_dir>/<group_id_hex>/<subdirectory>/`)
+- Store files atomically (write to `.tmp`, then rename)
+- Retrieve file paths
+- Check file existence
+- Find files by prefix (useful when extension is unknown)
+
+**Does NOT:**
+- Interact with the database
+- Download or upload files
+- Handle encryption/decryption
+- Know about Blossom URLs
+
+**Key Types:**
+- `MediaFileStorage` - Filesystem storage manager
+- `Storage` - Wrapper struct for all storage subsystems
+
+**Directory Structure:**
+```
+<data_dir>/media_cache/
+  <mls_group_id_hex>/
+    group_images/          # Group profile images (marmot extension)
+      <hash>.<ext>
+    media/                 # Chat media files
+      <hash>.<ext>
```

Comment:
same comment here regarding scoping the files on the fs. I think we can rely on the hash to deduplicate without needing to rely on directory structure